lib/proxy/log_buffer/decorator.rb

```diff
@@ -38,6 +38,14 @@ def add(severity, message = nil, progname = nil, backtrace = nil)
           message = progname
         end
       end
+      # accepts backtrace, exception and simple string for historical reasons
+      backtrace = if exception_or_backtrace.is_a?(Exception) && !exception_or_backtrace.backtrace.nil?
```

Comment:
I don't see why you're moving this out here. It's only used on line 57 so nesting it inside the if there should be good enough?

Reply:
Right, this was done for no particular reason. Rebased.